src/main/java/teamseven/ce216project/Library.java

```diff
@@ -1,10 +1,200 @@
+                foundBooks.add(book);
+                isFound = true;
+
+            }
+        }
+    }
+
+    public void searchForLanguage(String string, Book book){
+        if(book.getLanguage()!=null) {
+            if (book.getLanguage().contains(string) && !isFound) {
+                foundBooks.add(book);
+                isFound = true;
+
+            }
+        }
+    }
+
+    public void searchForDate(String string, Book book){
+        if(book.getDate()!=null) {
+            if (book.getDate().contains(string) && !isFound) {
+                foundBooks.add(book);
+                isFound = true;
+
+            }
+        }
+    }
+
+    public void searchForRating(String string, Book book){
+        if(book.getRating()!=null) {
+            float rating = Float.parseFloat(string);
+            float bookRating = Float.parseFloat(book.getRating());
+            if (bookRating == rating && !isFound) {
+                foundBooks.add(book);
+                isFound = true;
+            }
+        }
+    }
+
+    public void searchForEdition(String string, Book book){
+        if(book.getEdition()!=null) {
+            int edition = Integer.parseInt(string);
+            int bookEdition = Integer.parseInt(book.getEdition());
+            if (bookEdition == edition && !isFound) {
+                foundBooks.add(book);
+                isFound = true;
+            }
+        }
+    }
+
+    public void searchForNumberOfPages(String string, Book book){
+        if(book.getNumberOfPages()!=null) {
+            int numberOfPages = Integer.parseInt(string);
+            int bookPages = Integer.parseInt(book.getNumberOfPages());
+            if (bookPages == numberOfPages && !isFound) {
+                foundBooks.add(book);
+                isFound = true;
+            }
+        }
+    }
+
+    public void searchForAuthor(String string, Book book){
+        if(book.getAuthors()!=null){
+            for(int i = 0; i<book.getAuthors().size();i++) {
+                 if (book.getAuthors().get(i).contains(string) && !isFound) {
+                    foundBooks.add(book);
+                    isFound = true;
+            }
+          }
+        }
+    }
+
+    public void searchForTranslator(String string, Book book){
+        if(book.getTranslators()!=null) {
+            for (int i = 0; i < book.getTranslators().size(); i++) {
+                if (book.getTranslators().get(i).contains(string) && !isFound) {
+                    foundBooks.add(book);
+                    isFound = true;
+                }
+            }
+        }
+    }
+
+    //Tag Search
+    //and Management
+
+    public void addTag(Book book){
+        isFound=false;
+        for (String s: book.getTags() ){
+            for (String uniqueS : uniqueTags){
+                if(uniqueS.equals(s))isFound=true;
+            }
+            if(!isFound) {
+                uniqueTags.add(s);
+                break;
```

Comment:
- This break stops the loop on the line 173. It will make it so that whenever a tag is found, it will not continue check the other tags of the books. Please solve this issue buy not breaking. 

- isFound does not reset after checking all the uniqueTags.